units/patch_intent.go

```diff
@@ -535,6 +525,30 @@ func (j *patchIntentProcessor) setToPreviousPatchDefinition(patchDoc *patch.Patc
 	return previousPatch.Status, nil
 }
 
+// updatePatchDocPatches updates the commit sha for existing patch modules and adds new modules to the patch for modules
+// that exist in the project but not in the patch. This is necessary so that the manifest load endpoint will include the
+// updated commit sha in the response's module overrides field.
+func updatePatchDocPatches(ctx context.Context, patchDoc *patch.Patch, project *model.Project, token string) error {
+	for _, mod := range project.Modules {
+		sha, err := thirdparty.GetBranchCommitHash(ctx, mod.Repo, mod.Branch, token)
+		if err != nil {
+			return errors.Wrapf(err, "getting branch for module '%s'", mod.Branch)
+		}
+		for i := range patchDoc.Patches {
+			patchName := patchDoc.Patches[i].ModuleName
+			if patchName == mod.Name {
+				patchDoc.Patches[i].Githash = sha
+			}
+		}
+		patchDoc.Patches = append(patchDoc.Patches, patch.ModulePatch{
```

Comment:
This should only happen if we don't find the module name at 539, right? Should we add a continue up there?

Reply:
Yes 🤦‍♂️ I had meant to add some logic to enforce that but evidently not! Fixed now. I think a `found` variable is necessary instead of adding a continue though.